sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/trace/sampler/aws_xray_remote_sampler.py

```diff
@@ -0,0 +1,208 @@
+class _AwsXRayRemoteSampler(Sampler):
+    def __init__(
+        self,
+        resource: Resource,
+        endpoint: Optional[str] = None,
+        polling_interval: Optional[int] = None,
+        log_level: Optional[str] = None,
+    ):
+        self._root = ParentBased(
+            _InternalAwsXRayRemoteSampler(
+                resource=resource,
+                endpoint=endpoint,
+                polling_interval=polling_interval,
+                log_level=log_level,
+            )
+        )
+
+    @override
+    def should_sample(
+        self,
+        parent_context: Optional["Context"],
+        trace_id: int,
+        name: str,
+        kind: Optional[SpanKind] = None,
+        attributes: Optional[Attributes] = None,
+        links: Optional[Sequence["Link"]] = None,
+        trace_state: Optional["TraceState"] = None,
+    ) -> "SamplingResult":
+        return self._root.should_sample(
+            parent_context,
+            trace_id,
+            name,
+            kind=kind,
+            attributes=attributes,
+            links=links,
+            trace_state=trace_state,
+        )
+
+    # pylint: disable=no-self-use
+    @override
+    def get_description(self) -> str:
+        return f"AwsXRayRemoteSampler{{root:{self._root.get_description()}}}"
+
+
+# WORK IN PROGRESS
+#
+# _InternalAwsXRayRemoteSampler contains all core XRay Sampler Functionality,
+# however it is NOT Parent-based (e.g. Sample logic runs for each span)
+# Not intended for external use, use Parent-based `AwsXRayRemoteSampler` instead.
+class _InternalAwsXRayRemoteSampler(Sampler):
+    """
+    Remote Sampler for OpenTelemetry that gets sampling configurations from AWS X-Ray
+
+    Args:
+        resource: OpenTelemetry Resource (Required)
+        endpoint: proxy endpoint for AWS X-Ray Sampling (Optional)
+        polling_interval: Polling interval for getSamplingRules call (Optional)
+        log_level: custom log level configuration for remote sampler (Optional)
+    """
+
+    def __init__(
+        self,
+        resource: Resource,
+        endpoint: Optional[str] = None,
+        polling_interval: Optional[int] = None,
+        log_level: Optional[str] = None,
+    ):
+        # Override default log level
+        if log_level is not None:
+            _logger.setLevel(log_level)
+
+        if endpoint is None:
+            _logger.info(
+                "`endpoint` is `None`. Defaulting to %s",
+                DEFAULT_SAMPLING_PROXY_ENDPOINT,
+            )
+            endpoint = DEFAULT_SAMPLING_PROXY_ENDPOINT
+        if polling_interval is None or polling_interval < 10:
+            _logger.info(
+                "`polling_interval` is `None` or too small. Defaulting to %s",
+                DEFAULT_RULES_POLLING_INTERVAL_SECONDS,
+            )
+            polling_interval = DEFAULT_RULES_POLLING_INTERVAL_SECONDS
+
+        self.__client_id = self.__generate_client_id()  # pylint: disable=W0238
+        self._clock = _Clock()
```

Comment:
This is not used right?

Reply:
Right. I've removed it in latest commit, I'll add it back when it's needed.